src/FirebaseArray.js

```diff
@@ -109,12 +109,25 @@
          */
         $add: function(data) {
           this._assertNotDestroyed('$add');
+          var self = this;
           var def = $firebaseUtils.defer();
-          var ref = this.$ref().ref().push();
-          ref.set($firebaseUtils.toJSON(data), $firebaseUtils.makeNodeResolver(def));
-          return def.promise.then(function() {
-            return ref;
-          });
+          var ref = this.$ref().ref.push();
+          var dataJSON;
+
+          try {
+            dataJSON = $firebaseUtils.toJSON(data);
+          } catch (err) {
+            def.reject(err);
+          }
+
+          if (dataJSON) {
```

Comment:
This check technically changes behavior unless we can prove that `dataJSON` is never empty. I'm not sure if we actually want the check here. For example, if `dataJSON` is `{}`, we should probably allow that write (which will essentially do a remove).